File: server/gb_api/api_objects/video_category.py
from . import image
from .api_object import GuidlessApiObject, GuidlessApiObjectDefinition, ApiField


class VideoCategoryDefinition(GuidlessApiObjectDefinition):
    def __init__(self):
        item_name = 'video_category'
        collection_name = 'video_categories'
        type_id = 2320
        object_class = VideoCategory
        fields = [
            # URL pointing to the video_category detail resource.
            ApiField('api_detail_url', str),
            # Brief summary of the video_category.
            ApiField('deck', str),
            # Unique ID of the video_category.
            ApiField('id', int),
            # Name of the video_category.
            ApiField('name', str),
            # Main image of the video_category.
            ApiField('image', image.Image),
            # URL pointing to the video_category on Giant Bomb.
            ApiField('site_detail_url', str)
        ]
        super().__init__(item_name, collection_name, type_id, object_class, fields)


class VideoCategory(GuidlessApiObject):
    __definition = None

    def __init__(self, result):
        if self.__definition is None:
            self.__definition = VideoCategoryDefinition()
        super().__init__(self.__definition, result)

File: server/app/core/ts/VideoCategory.ts
import Loadable from "./Loadable";
import Definitions from "./Definitions";
import {ImageResponseData} from "./gbdlapi/ImagesAPI";
import API, {validResponseData} from "./gbdlapi/API";
import VideoShowsAPI, {VideoShowResponseData} from "./gbdlapi/VideoShowsAPI";
import Video from "./Video";
import ResultList from "./ResultList";
import {VideoCategoryResponseData} from "./gbdlapi/VideoCategoriesAPI";

export default class VideoCategory extends Loadable {
    /** URL pointing to the video_category detail resource. */
    apiDetailUrl: string
    /** Brief summary of the video_category. */
    deck: string
    /** Unique ID of the video_category. */
    id: number
    /** Name of the video_category. */
    name: string
    imageId: number
    /** Main image of the video_category. */
    image: ImageResponseData
    /** URL pointing to the video_category on Giant Bomb. */
    siteDetailUrl: string

    public constructor(data?: VideoCategoryResponseData) {
        super();
        if (!validResponseData(data)) {
            this.loadFailed = true;
            return;
        }
        this.apiDetailUrl = data.api_detail_url;
        this.deck = data.deck;
        this.id = data.id;
        this.name = data.name;
        this.imageId = data.image_id;
        this.image = data.image ?? null;
        this.siteDetailUrl = data.site_detail_url;
    }

    public static async getAll() {
        let response = await API.videoCategories.getAll();
        return new ResultList(VideoCategory, response.data);
    }
}